src/backend/distributed/commands/function.c

```diff
@@ -638,6 +637,19 @@ UpdateFunctionDistributionInfo(const ObjectAddress *distAddress,
 	systable_endscan(scanDescriptor);
 
 	table_close(pgDistObjectRel, NoLock);
+
+	if (EnableDependencyCreation)
```

Comment:
We should add a test to show this works as expected:
```SQL
-- make sure that metatadata sync is possible
set citus.replication_model TO streaming ;


CREATE TABLE test (a int, b int);
SELECT create_distributed_table('test', 'a');


create or replace procedure proc(dist_key integer, dist_key_2 integer)
language plpgsql    
as $$
DECLARE
    res INT := 0;
begin
    INSERT INTO test VALUES (dist_key);
    SELECT count(*) INTO res FROM test;
    RAISE NOTICE 'Res: %', res;
    commit;
end;$$;

SELECT create_distributed_function('proc(integer, integer)', 'dist_key', 'test' );
 SELECT distribution_argument_index FROM citus.pg_dist_object WHERE objid = 'proc'::regproc;
SELECT run_command_on_workers($$ SELECT distribution_argument_index FROM citus.pg_dist_object WHERE objid = 'proc'::regproc;$$);

-- make sure that metadata is synced, it may take few seconds
SELECT bool_and(hasmetadata) FROM pg_dist_node;

-- pushdown the procedure to the node which does a multi-shard query
CALL proc(1);


-- re-distribute and show that now the distribution_argument_index is updated on both the coordinator and workers
SELECT create_distributed_function('proc(integer, integer)', 'dist_key_2', 'test' );
SELECT * FROM citus.pg_dist_object WHERE objid = 'proc'::regproc;
 SELECT distribution_argument_index FROM citus.pg_dist_object WHERE objid = 'proc'::regproc;

SELECT run_command_on_workers($$ SELECT distribution_argument_index FROM citus.pg_dist_object WHERE objid = 'proc'::regproc;$$);
```